cpp/src/command_classes/CentralScene.cpp

```diff
@@ -201,10 +234,52 @@ bool CentralScene::HandleMsg
 
 		if( Node* node = GetNodeUnsafe() )
 		{
-				char lbl[64];
-				for (int i = 1; i <= m_scenecount; i++) {
-					snprintf(lbl, 64, "Scene %d", i);
-					node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), _instance, i, lbl, "", true, false, 0, 0 );
+				for (int i = 1; i <= m_scenecount && i <= 31; i++) {
+					if ( version2orHigher )
+					{
+					    if ( identical )
+					    {
+					        // instance for scene number normaly isn't right, but in this case i think this is the best solution.
+					        // Because i higly doubt that someone will ever use the multichannel class together with the centralScene class.
+                            int keyAttributes = _data[3];
+                            if ( keyAttributes & CentralSceneMask_KeyPressed1time)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_SinglePress, "Scene", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_KeyPressed2times)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_DoublePress, "Scene double press", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_HeldDown)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_HeldDown, "Scene held down", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_KeyReleased)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_Released, "Scene released", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_KeyPressed3times)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_TriplePress, "Scene triple press", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_KeyPressed4times)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_Pressed4Times, "Scene pressed 4 times", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_KeyPressed5times)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_Pressed5Times, "Scene pressed 5 times", "", true, false, 0, 0 );
+                            }
+                        }
```

Comment:
Yeah... I was thinking about this as well. 
My first thought was for devices that support Identical Capabilities, Delay creating the list ValueID till you get a SupportedReport message, and "filter" this list with only supported Key Attributes.

Then from a application, once you get a ValueAdded Notification for this list, you can iterate over the list.

But for non-identical capabilities for Scenes, this breaks..... 

I thought just assign a high index number (say 100) and use a List ValueID for each Scene (so index is 100+Scene Number), but this would break as we can have upto 255 Scenes, which would overflow our index (and IMHO, isn't a great approach anyway).

let me sleep on this....